File: src/main/scala/za/co/absa/statusboard/model/RefinedStatus.scala
package za.co.absa.statusboard.model

import io.circe.generic.JsonCodec

import java.time.Instant

@JsonCodec
case class RefinedStatus(
  serviceName: String,
  env: String,
  status: RawStatus,
  maintenanceMessage: String,
  firstSeen: java.time.Instant = Instant.now(),
  lastSeen: java.time.Instant,
  notificationSent: Boolean
)

File: ui/src/app/models/refined-status.ts
import { RawStatus } from './raw-status';

export interface RefinedStatus {
  serviceName: string;
  env: string;
  status: RawStatus;
  maintenanceMessage: string;
  firstSeen: string; // format: date-time
  lastSeen: string; // format: date-time
  notificationSent: boolean;
}
